js/visearch.js

```diff
@@ -195,18 +198,48 @@
   };
 
   prototypes.product_search_by_image = function (params, options, callback, failure) {
+    let altOptions;
+    let altCallback;
+    if (isFunction(options)) {
+      altOptions = callbackWrap.bind(this, options);
+      altCallback = callback;
+    } else {
+      altOptions = options;
+      altCallback = callbackWrap.bind(this, callback);
+    }
+
     return productsearch.searchbyimage(params, getDefaultTrackingParams(),
-      options, callbackWrap.bind(this, callback), failure);
+      altOptions, altCallback, failure);
   };
 
   prototypes.product_search_by_id = function (productId, params, options, callback, failure) {
+    let altOptions;
+    let altCallback;
+    if (isFunction(options)) {
+      altOptions = callbackWrap.bind(this, options);
+      altCallback = callback;
+    } else {
+      altOptions = options;
+      altCallback = callbackWrap.bind(this, callback);
+    }
+
     return productsearch.searchbyid(productId, params, getDefaultTrackingParams(),
-      options, callbackWrap.bind(this, callback), failure);
+      altOptions, altCallback, failure);
   };
 
   prototypes.product_recommendations = function (productId, params, options, callback, failure) {
+    let altOptions;
```

Comment:
no need to duplicate the code here, just call `product_search_by_id`, they are exactly the same